mops/src/hpe/cpu.tpp

```diff
@@ -31,8 +31,8 @@ void _homogeneous_polynomial_evaluation_templated_polynomial_order(
     size_t size_first_dimension_interleft = size_first_dimension / simd_element_count;
     size_t size_remainder = size_first_dimension % simd_element_count;
 
-    scalar_t* interleft_a_ptr = new scalar_t[size_first_dimension_interleft*n_possible_factors*simd_element_count];
-    scalar_t* remainder_a_ptr = new scalar_t[size_remainder*n_possible_factors];
+    scalar_t* interleft_a_ptr = std::vector<scalar_t>(size_first_dimension_interleft*n_possible_factors*simd_element_count).data();
```

Comment:
This is giving you a dangling pointer. 

```suggestion
    auto interleft_a = std::vector<scalar_t>(size_first_dimension_interleft*n_possible_factors*simd_element_count);
    scalar_t* interleft_a_ptr = interleft_a.data();
```

Given the usage in the code, I think we can even directly use the vector 

```suggestion
    auto interleft_a = std::vector<scalar_t>(size_first_dimension_interleft*n_possible_factors*simd_element_count);
```